### DISCUSSION TITLE
Bulk upload users csv

### DISCUSSION THREAD

**Discussion by andyburgessmd** [Budibase/budibase#7397] [Help] [answered]
Has anyone found any documentation or worked out the format of the csv that can be used to add new users though the portal?

**Comment by R2bEEaton** [comment, marked as answer]
Hi @andyburgessmd,

There currently isn't documentation for how to import users via CSV. Looking at the code and the GUI modal, I figured out that you can only import emails (no names or anything else). You can also set the role of all the emails of entire CSV file using the GUI, but not assign apps. It looks like you will be able to do so in the future with Groups.

However, when I created a CSV using Microsoft Excel with emails on their own lines (as the code implies), I receive the following message.
```
Error, please check the following emails: example1@test.com , example2@test.com
```
I got the same error when using Notepad++. It appeared that there was a trailing space after each email even though I didn't include one. Then I used Notepad++ to replace `\r` with an empty string, and reuploaded the CSV. This time it worked. So please note that if you make this CSV file on Windows, your newline will include a Carriage Return character (`\r`) and that the Budibase file parser does not like it. I will create an issue for the problem, but in the mean time, know the following:

1. To import users, create a file containing list of valid emails (each on their own line) and save it with a `.csv` extension.
2. If using Windows or a text editor that includes the carriage return, use Notepad++ or a similar program to remove `\r`.
3. Upload the CSV to Budibase and select the Budibase role you want to import all the users as.

Hope this helps!

(The problematic code is [here](https://github.com/Budibase/budibase/blob/f2bfe40c60317f75583371205a78769508ca955e/packages/builder/src/pages/builder/portal/manage/users/_components/ImportUsersModal.svelte#L65))
(I also created a PR that should address this #7399)
